avocado/core/nrunner.py

```diff
@@ -356,9 +356,37 @@ class BaseRunner(abc.ABC):
     Base interface for a Runner
     """
 
+    @staticmethod
+    def _get_avocado_version():
+        """Return the Avocado package version, if installed"""
+        version = "unknown.unknown"
+        if PKG_RESOURCES_AVAILABLE:
+            try:
+                version = pkg_resources.get_distribution(
+                    'avocado-framework').version
+            except pkg_resources.DistributionNotFound:
+                pass
+        return version
+
     def __init__(self, runnable):
         self.runnable = runnable
 
+        # set default Avocado env variables
+        #
+        # NOTE: these environment variables are redefined inside an Avocado
+        # instrumented test.
+        #
+        # this is a code duplication from avocado/core/version.py to avoid
+        # nrunner depending on other than python modules
+        os.environ['AVOCADO_VERSION'] = self._get_avocado_version()
+
+        # this is a code duplication from avocado/core/data_dir.py to avoid
+        # nrunner depending on other than python modules
+        os.environ['AVOCADO_TEST_BASEDIR'] = self.runnable.config.get(
```

Comment:
Hi @willianrampazzo, I think in here we should validate here if the directories exist and can be used in the runner environment. Because the environment of runner can be different from environment of avocado core. Or maybe we can limit this just for the `ProcessSpawner` where the environments are the same. What do you think?

Reply:
About the verification, as far as I could check, the directories are always available, but on the local host, where Avocado is executed.

On the other hand, as you mentioned, some may not make sense as they are located in the host and not in the spawner environment. This was something I was expecting to discuss on https://github.com/avocado-framework/avocado/discussions/4751 and is a valid point.

So far, for me, it looks like we are porting something from the runner architecture that does not make much sense on the nrunner architecture, at least when porting it 1 to 1.

Reply:
So maybe for now we can create warning in documentation that this can be used only with the `ProcessSpawner`.